heat/fft/fft.py

```diff
@@ -0,0 +1,787 @@
+    del _
+    # redistribute partial result from axis 1 to 0
+    partial_ht_result.resplit_(axis=0)
+    if original_split != 0:
+        # transpose x, partial_ht_result back to original shape
+        x = x.transpose(transpose_axes)
+        partial_ht_result = partial_ht_result.transpose(transpose_axes)
+
+    # now apply FFT along leftover (non-split) axes
+    axes = list(axes)
+    axes.remove(original_split)
+    axes = tuple(axes)
+    if s is not None:
+        s = list(s)
+        s = s[:split_index] + s[split_index + 1 :]
+        s = tuple(s)
+    ht_result = __fftn_op(partial_ht_result, fftn_op, s=s, axes=axes, norm=norm)
+    del partial_ht_result
+    if real_op:
+        # discard elements beyond Nyquist frequency on last transformed axis
+        nyquist_slice = [slice(None)] * ht_result.ndim
+        nyquist_slice[axes[-1]] = slice(0, nyquist_freq)
+        ht_result = ht_result[(nyquist_slice)].balance_()
+    return ht_result
+
+
+def __real_fft_op(x: DNDarray, fft_op: callable, **kwargs) -> DNDarray:
+    if x.larray.is_complex():
+        raise TypeError(f"Input array must be real, is {x.dtype}.")
+    return __fft_op(x, fft_op, **kwargs)
+
+
+def __real_fftn_op(x: DNDarray, fftn_op: callable, **kwargs) -> DNDarray:
+    if x.larray.is_complex():
+        raise TypeError(f"Input array must be real, is {x.dtype}.")
+    return __fftn_op(x, fftn_op, **kwargs)
+
+
+def fft(x: DNDarray, n: int = None, axis: int = -1, norm: str = None) -> DNDarray:
+    """
+    Compute the one-dimensional discrete Fourier Transform.
+
+    This function computes the one-dimensional discrete Fourier Transform over the specified axis in an M-dimensional
+    array by means of the Fast Fourier Transform (FFT). By default, the last axis is transformed, while the remaining
+    axes are left unchanged.
+
+    Parameters
+    ----------
+    x : DNDarray
+        Input array, can be complex. WARNING: If x is 1-D and distributed, the entire array is copied on each MPI process.
+    n : int, optional
+        Length of the transformed axis of the output. If not given, the length is taken to be the length of the input
+        along the axis specified by axis. If `n` is smaller than the length of the input, the input is cropped. If `n` is
+        larger, the input is padded with zeros. Default: None.
+    axis : int, optional
+        Axis over which to compute the FFT. If not given, the last axis is used, or the only axis if x has only one
+        dimension. Default: -1.
+    norm : str, optional
+        Normalization mode: 'forward', 'backward', or 'ortho' (see `numpy.fft` for details). Default is "backward".
+
+    Notes
+    -----
+    This function requires MPI communication if the input array is transformed along the distribution axis.
+    If the input array is 1-D and distributed, this function copies the entire array on each MPI process! i.e. if the array is very large, you might run out of memory.
```

Comment:
From my point of view, this is OK. However, I would recommend to open a corresponding issue to work on an alternative specifically for this distributed 1d-case (or does this not make any sense?)

Reply:
Created issue #1263